lib/pbench/server/api/resources/users_api.py

```diff
@@ -269,14 +268,24 @@ def post(self):
                         "message": "failure message"
                     }
         """
-        auth_token = self.auth.token_auth.get_auth()["token"]
-        user = self.auth.token_auth.current_user()
+        auth_token = self.auth.get_auth_token(self.logger)
+        user = Auth.verify_auth(auth_token=auth_token)
+
+        # "None" user represents that either the token is not present in our database or it is an expired token.
+        # Expired token is already deleted by now if we reach here.
+        # In either case we will return suceess to a client, since the same token can not be used again.
 
         try:
             ActiveTokens.delete(auth_token)
-            self.logger.info(
-                "Auth token entry deleted for user with username {}", user.email
-            )
+            # Do we need to log every time someone logs out
+            if user:
+                self.logger.info(
+                    "Auth token entry deleted for user with email {}", user.email
```

Comment:
... but I think I'd prefer directly saying that the user logged out rather than reporting in terms of the auth token.

Also, reporting the username is probably more useful right now. If we drop that in favor of email (of which I'm not in favor ;-) ), we'll figure out what to do then.

Reply:
On the server side username is fine to report in the logs.  The use of "email" to login is a user-interface thing, not an implementation thing.